000-mm2-connector-offsets-support.md

```diff
@@ -0,0 +1,449 @@
+
+The response body received when using the GET endpoint, and the request body used when calling the PATCH endpoint have identical structures to allow users to easily fetch and then alter offsets.
+The structure of the body is not fully defined for generic source connectors.
+This is because source connectors determine their own structure for offsets depending on the source system they are integrating with.
+However, for each of the MirrorMaker connectors (MirrorSourceConnector, MirrorCheckpointConnector, MirrorHeartbeatConnector) we know the expected structure.
+
+MirrorSourceConnector format:
+```
+{
+  "offsets": [
+    {
+      "partition": {
+        "cluster": "east-kafka",
+        "partition": 0,
+        "topic": "mirrormaker2-cluster-configs"
+      },
+      "offset": {
+        "offset": 0
+      }
+    }
+  ]
+}
+```
+
+MirrorCheckpointConnector format:
+```
+{
+  "offsets": [
+    {
+      "partition": {
+        "partition": 4,
+        "topic": "inventory",
+        "group": "my-group"
+      },
+      "offset": {
+        "offset": 0
+      }
+    }
+  ]
+}
+```
+
+MirrorHeartbeatConnector format:
+```
+{
+  "offsets": [
+    {
+      "partition": {
+        "sourceClusterAlias": "east-kafka",
+        "targetClusterAlias": "west-kafka"
+      },
+      "offset": {
+        "offset": 0
+      }
+    }
+  ]
+}
+```
+
+### New annotations
+
+All three actions will make use of a new annotation called `strimzi.io/connector-offsets`.
+The possible values will be `list`, `alter`, and `reset`.
+
+A second new annotation will be added called `strimzi.io/mirrormaker-connector`.
+It will be required when `strimzi.io/connector-offsets` is set.
+The value will be the name of the connector to apply the action to, for example `east-kafka->west-kafka.MirrorSourceConnector`.
+
+If the `strimzi.io/mirrormaker-connector` annotation is set, but the `strimzi.io/connector-offsets` annotation is missing the operator will take no action but add a warning message.
+
+### Listing offsets
+
+To list the current offsets the user will annotate the KafkaMirrorMaker2 resource with the new annotation `strimzi.io/connector-offsets` set to `list`, and the `strimzi.io/mirrormaker-connector` annotation.
+After the annotations are added, on the next reconciliation the operator will fetch the current offsets for the connector and create a Kubernetes ConfigMap containing the JSON response.
+The operator will add an owner reference to the Kubernetes ConfigMap pointing to the KafkaMirrorMaker2 resource that was annotated.
+If the ConfigMap already exists the operator will patch the ConfigMap with updated data and not add an owner reference.
+This means any existing keys in the ConfigMap that the operator is not updating will remain as before.
+Once all list operation is complete the operator will then remove both the `strimzi.io/mirrormaker-connector` and `strimzi.io/connector-offsets` annotations from the KafkaMirrorMaker2 CR.
```

Comment:
nit: `all list operations are`?

Reply:
This should be `Once the list operation is`, since I changed the proposal to only accept one connector at a time.